new/postinstall.sh

```diff
@@ -1,9 +1,19 @@
 #!/bin/bash
 set -ex
 
+# Force message cache to populate
+php $PATCHDEMO/wikis/$NAME/w/maintenance/language/alltrans.php > /dev/null
+
+# Fetch main page title from message cache
+# TODO: This would be much neater if we just had a maintenance script that could access Title:newMainPage.
+MAINPAGETITLE=$( mysql \
+	-u patchdemo -ppatchdemo patchdemo_$NAME -B -N -e "SELECT lc_value FROM l10n_cache where lc_key = 'messages:mainpage' LIMIT 1" | \
+	php -r "echo unserialize(file_get_contents('php://stdin'));" \
+)
+
```

Comment:
This is horrific, but I couldn't find a better way.

We can patch maintenance/edit.php to support editing the main page, but that will never work with old versions of core, although perhaps that's ok...

Reply:
See https://gerrit.wikimedia.org/r/c/mediawiki/core/+/739555

Reply:
I was going to suggest this:
```
echo "{{int:mainpage}}" | php maintenance/parse.php
```
…but, that actually outputs HTML wrapped in `<p>…</p>`, so you'd need some more weird stuff to remove that.

Or, maybe we can just query api.php?action=query&format=json&meta=siteinfo and parse the "mainpage" given there?

Reply:
`{{int:mainpage}}` seems a little neater.

Reply:
Done. I've left the comment about fixing edit.php in because I think that is the best solution.

Reply:
What about:
```
echo 'echo Title::newMainPage()->getPrefixedText()' | php maintenance/eval.php 
Main Page
```
?

Reply:
Thanks, I tried eval.php but for some reason couldn't get it to read stdin.

In your example the space in "Main Page" causes it to write to the article "Main", but that should be fixable.